include/kokkos-utils/callbacks/SequenceOfRegionTimer.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_SEQUENCEOFREGIONTIMER_HPP
+#define KOKKOS_UTILS_CALLBACKS_SEQUENCEOFREGIONTIMER_HPP
+
+#include "kokkos-utils/callbacks/RegionTimerListener.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+/**
+ * @brief Time a sequential series of push/pop regions.
+ *
+ * The regions must not be interleaved.
+ *
+ * @note Overhead should be as low as possible.
+ */
+template <Matcher MatcherType>
+struct SequenceOfRegionTimer
```

Comment:
We could try to find something in the name to make it different from our other "timers". Like `SequenceOfRegionTimerManaged`, `SequenceOfRegionTimerTool`, ...